Sources/SentrySwiftUI/SentryTracedView.swift

```diff
@@ -38,62 +109,56 @@ import SwiftUI
+        let name = viewName ?? SentryTracedView.extractName(content: Content.self)
+        let nameSource = viewName == nil ? SentryTransactionNameSource.component : SentryTransactionNameSource.custom
+        let waitforFullDisplay = waitForFullDisplay ?? SentrySDK.options?.enableTimeToFullDisplayTracing ?? false
+        self.viewModel = SentryTraceViewModel(name: name, nameSource: nameSource, waitForFullDisplay: waitforFullDisplay)
     }
-
+#else
+    /// Creates a view that measures the performance of its `content`.
+    ///
+    /// - Parameter viewName: The name that will be used for the span, if nil we try to get the name of the content class.
+    /// - Parameter content: The content that you want to track the performance
+    public init(_ viewName: String? = nil, @ViewBuilder content: @escaping () -> Content) {
+        self.content = content
+        let name = viewName ?? SentryTracedView.extractName(content: Content.self)
+        let nameSource = viewName == nil ? SentryTransactionNameSource.component : SentryTransactionNameSource.custom
+        self.viewModel = SentryTraceViewModel(name: name, nameSource: nameSource, waitForFullDisplay: false)
+    }
+#endif
+    
     private static func extractName(content: Any) -> String {
         var result = String(describing: content)
-
+        
         if let index = result.firstIndex(of: "<") {
             result = String(result[result.startIndex ..< index])
         }
-
+        
         return result
     }
-
+    
     public var body: some View {
-        if viewAppeared {
-            return self.content().onAppear()
-        }
-
-        var transactionCreated = false
-        if SentryPerformanceTracker.shared.activeSpanId() == nil {
-            transactionCreated = true
-            let transactionId = SentryPerformanceTracker.shared.startSpan(withName: self.name, nameSource: nameSource, operation: "ui.load", origin: self.traceOrigin)
-            SentryPerformanceTracker.shared.pushActiveSpan(transactionId)
-
-            //According to Apple's documentation, the call to `body` needs to be fast
-            //and can be made many times in one frame. Therefore they don't use async code to process the view.
-            //Scheduling to finish the transaction at the end of the main loop seems the least hack solution right now.
-            //'onAppear' is not a suitable place to do this because it may happen before other view `body` property get called.
-            DispatchQueue.main.async {
-                SentryPerformanceTracker.shared.popActiveSpan()
-                SentryPerformanceTracker.shared.finishSpan(transactionId)
-            }
-        }
-
-        let id = SentryPerformanceTracker.shared.startSpan(withName: transactionCreated ? "\(self.name) - body" : self.name, nameSource: nameSource, operation: "ui.load", origin: self.traceOrigin)
-
-        SentryPerformanceTracker.shared.pushActiveSpan(id)
+        let spanId = viewModel.startSpan()
```

Comment:
`m:` Would this create a new span on every single re-calculation of the SwiftUI view hierarchy, instead of only the first time?
Do we know if this has a performance impact?

Reply:
No, it is only created at the first time. The `startSpan()` function checks for that.